src/signal.rs

```diff
@@ -41,13 +42,13 @@ fn into_raw<F: FnMut() -> Continue + 'static>(func: F) -> gpointer {
 ///
 /// Similar to `glib::idle_add` but only callable from the main thread and
 /// doesn't require `Send`.
-pub fn idle_add<F>(func: F) -> u32
+pub fn idle_add<F>(func: F) -> SourceId
     where F: FnMut() -> Continue + 'static {
     assert_initialized_main_thread!();
-    unsafe {
+    from_glib(unsafe {
```

Comment:
I thought that since the compiler is happy with it, it is better to have it out of the unsafe {} block. Will move it inside.